foodji_api/DomainTests/Recipes/RecipeTests.cs

```diff
@@ -54,6 +80,58 @@ public void GivenValidValues_Create_ReturnsRecipe()
             actualRecipe.Author.Should().Be(expectedAuthor);
             actualRecipe.Should().BeOfType<Recipe>();
         }
+
+        [Test]
+        public void GivenValidRecipeToUpdate_UpdateRecipe_RecipeUpdated()
+        {
+            // Arrange new substitute
+            var substitute = RecipeSubstitute.Create(
+                "updatedSubstituteName",
+                "updatedSubstituteDescription",
+                "substituteDescription",
+                Measurement.Create(
+                    1,
+                    UnitType.Millilitre,
+                    "updated alternative text"),
+                new List<Tag> { Tag.Vegan });
+                
+            // Arrange new ingredient
+            var ingredient = RecipeIngredient.Create(
+                "updatedIngredientName",
+                "updatedIngredientDescription",
+                Measurement.Create(
+                    2,
+                    UnitType.Gram,
+                    "updated alternative text"),
+                new List<Tag> { Tag.Vegan },
+                new List<RecipeSubstitute> { substitute });
+
+            // Arrange new recipe
+            var expectedRecipe = Recipe.Create(
+                "expectedName",
+                RecipeCategory.Dessert,
+                "expectedDescription",
+                RecipeDetails.Create(1,1,1,1),
+                new List<RecipeIngredient> { ingredient },
+                new List<string> { "updatedRecipeStep" },
+                new Uri("https://www.yahoo.ca"),
+                "updatedAuthor");
+            
+            // Act
+            _recipe.Update(expectedRecipe);
+            
+            // Assert
+            // I know we should be testing the Id here,
+            // but I'm not sure how to do that since Create() does not instantiate the Id property...
+            _recipe.Name.Should().Be(expectedRecipe.Name);
+            _recipe.Category.Should().Be(expectedRecipe.Category);
+            _recipe.Description.Should().Be(expectedRecipe.Description);
+            _recipe.Details.Should().Be(expectedRecipe.Details);
+            _recipe.Ingredients.Should().BeEquivalentTo(expectedRecipe.Ingredients);
+            _recipe.Steps.Should().BeEquivalentTo(expectedRecipe.Steps);
+            _recipe.ImageUri.Should().Be(expectedRecipe.ImageUri);
+            _recipe.Author.Should().NotBe(expectedRecipe.Author);
```

Comment:
J'aime pas tant un `NotBe` avec un expected - à ce moment-là, ce n'est *pas* le résultat expected, mais bien seulement la *updated* recipe. J'irais avec updatedRecipe plutôt qu'expectedRecipe, et je ferais un check explicit aussi que c'est le même author qu'avant - le stocker dans une variable expectedAuthor indépendante.

Un peu clunky, mais c'est jsutement de quoi qui découle de la création entière d'une recette pour l'update, plutôt que de quoi de plus lightweight.

Reply:
Sounds good! Juste pour être clair, j'avais rien contre le `NotBe` en tant que tel, juste la contradiction syntaxique de "should not be (the) expected value". Mais avec les changements que tu mentionnes, ça me semble bon, et on se sauve du problème.